hersh/hersh.cpp

```diff
@@ -6,23 +6,26 @@
 
 using namespace hersh;
 
-const std::string defaultPrompt = "hersh";
+const std::string defaultPrompt = "hersh$ ";
 
 bool isEOF() {
   return std::cin.eof();
 }
 
 int main(int argc, char *argv[]) {
+  Reader reader;
```

Comment:
Should probably be initialized with a prompt. A `reader` without a prompt is like a chimney sweep without a chimney sweeper.

Reply:
I set Reader's default prompt to "hersh$ ". So no need to initialize. Or is that what you meant?

Reply:
Partly. It's better now. But, it doesn't really make sense to set the prompt after initializing a `Reader` so I would get rid of the setter and add a constructor that allows initializing it.

Reply:
I haven't read the whole sh specifications, but if it's similar to the default bash prompt behavior, the prompt changes each time you change directory. So I think we'll have to change the prompt a lot.

Reply:
👍